src/test/kotlin/graphics/scenery/tests/examples/volumes/ClientApplication.kt

```diff
@@ -0,0 +1,509 @@
+package graphics.scenery.tests.examples.volumes
+
+import graphics.scenery.*
+import graphics.scenery.backends.Renderer
+import graphics.scenery.backends.Shaders
+import graphics.scenery.compute.ComputeMetadata
+import graphics.scenery.compute.InvocationType
+import graphics.scenery.controls.TrackedStereoGlasses
+import graphics.scenery.textures.Texture
+import graphics.scenery.textures.UpdatableTexture
+import graphics.scenery.ui.SwingBridgeFrame
+import graphics.scenery.utils.DataCompressor
+import graphics.scenery.utils.Image
+import graphics.scenery.utils.VideoDecoder
+import graphics.scenery.volumes.DummyVolume
+import graphics.scenery.volumes.EmptyNode
+import graphics.scenery.volumes.TransferFunction
+import graphics.scenery.volumes.TransferFunctionEditor
+import graphics.scenery.volumes.vdi.VDIData
+import graphics.scenery.volumes.vdi.VDIDataIO
+import graphics.scenery.volumes.vdi.VDINode
+import net.imglib2.type.numeric.integer.UnsignedIntType
+import net.imglib2.type.numeric.real.FloatType
+import org.joml.Matrix4f
+import org.joml.Vector3f
+import org.joml.Vector3i
+import org.lwjgl.system.MemoryUtil
+import org.zeromq.SocketType
+import org.zeromq.ZContext
+import org.zeromq.ZMQ
+import org.zeromq.ZMQException
+import java.io.ByteArrayInputStream
+import java.nio.ByteBuffer
+import java.util.concurrent.atomic.AtomicInteger
+import kotlin.concurrent.thread
+import kotlin.system.measureNanoTime
+
+class ClientApplication : SceneryBase("Client Application", 512, 512)  {
+
+    var hmd: TrackedStereoGlasses? = null
+    var buffer: ByteBuffer = ByteBuffer.allocateDirect(0)
+    val context = ZContext(4)
+
+    val numSupersegments = 20
+    val skipEmpty = true
+    var vdiStreaming = true
+    var newVDI = false
+    var firstVDI = true
+    var firstVDIStream = true
+    var firstVR = true
+    var currentlyVolumeRendering = false
+    var videoDecoding = true
+
+    val compute = VDINode()
+    val switch = EmptyNode()
+
+    val vulkanProjectionFix =
+        Matrix4f(
+            1.0f, 0.0f, 0.0f, 0.0f,
+            0.0f, -1.0f, 0.0f, 0.0f,
+            0.0f, 0.0f, 0.5f, 0.0f,
+            0.0f, 0.0f, 0.5f, 1.0f)
+    fun Matrix4f.applyVulkanCoordinateSystem(): Matrix4f {
```

Comment:
Now that this function is used in multiple places, it should likely be moved into one of the helper classes in `utils`.